samples/boards/bbc_microbit/pong/src/main.c

```diff
@@ -489,26 +492,32 @@ void pong_remote_lost(void)
 static void configure_buttons(void)
 {
 	static struct gpio_callback button_cb_data;
-	const struct device *gpio;
 
-	gpio = device_get_binding(DT_GPIO_LABEL(DT_ALIAS(sw0), gpios));
+	if (!device_is_ready(sw0_gpio.port)) {
+		printk("%s: device not ready.\n", sw0_gpio.port->name);
+		return;
+	}
+
+	if (!device_is_ready(sw1_gpio.port)) {
+		printk("%s: device not ready.\n", sw1_gpio.port->name);
+		return;
+	}
 
-	gpio_pin_configure(gpio, DT_GPIO_PIN(DT_ALIAS(sw0), gpios),
-			   DT_GPIO_FLAGS(DT_ALIAS(sw0), gpios) | GPIO_INPUT);
-	gpio_pin_configure(gpio, DT_GPIO_PIN(DT_ALIAS(sw1), gpios),
-			   DT_GPIO_FLAGS(DT_ALIAS(sw1), gpios) | GPIO_INPUT);
+	if (sw0_gpio.port != sw1_gpio.port) {
```

Comment:
This could be checked with a build assertion.

Reply:
There commits where not meant for this PR, will address this over in https://github.com/zephyrproject-rtos/zephyr/pull/47430

Reply:
Not sure I see how to make this a build assert

Reply:
Will move the discussion on the microbit changes over to #47430